iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/symbols/ListSymbol.java

```diff
@@ -58,12 +60,33 @@ public ListSymbol reportItemIf(Predicate<ExpressionTree> predicate, String messa
     return this;
   }
 
+  public ListSymbol reportIfEmpty(String message, SecondaryLocation... secondaryLocations) {
+    if (isEmpty()) {
+      report(message, secondaryLocations);
+    }
+    return this;
+  }
+
   public Stream<ExpressionTree> getItemIf(Predicate<ExpressionTree> predicate) {
     return items.stream().filter(predicate);
   }
 
   public boolean isEmpty() {
-    return tree != null && items.isEmpty();
+    return tree != null && items.isEmpty() && !isByReference();
+  }
+
+  @Override
+  public boolean isPresent() {
+    return tree != null && !isByReference();
```

Comment:
I don't understand the test on `isByReference`.

Reply:
A list is the value of an argument. Argument values can also be defined via reference. So that there are no false positives, should one decide on the absence of a list, it must also be checked whether the list is not included by a reference.

Reply:
My concern is that the API is not obvious.
For example, it seems that `isPresent` and `reportIfAbsent` are not consistent for a `ListSymbol`.
It's not a big deal for now, but I think it's a sign that something should be refined.